Sources/Migration/DataPipelineMigrationAssistant.swift

```diff
@@ -8,7 +8,7 @@ public protocol DataPipelineMigrationAction: AutoMockable {
     func processEventFromBGQ(identifier: String, name: String, timestamp: String?, properties: [String: Any])
     func processDeleteTokenFromBGQ(identifier: String, token: String, timestamp: String)
     func processRegisterDeviceFromBGQ(identifier: String, token: String, timestamp: String, attributes: [String: Any]?)
-    func processPushMetricsFromBGQ(token: String, event: Metric, deliveryId: String, timestamp: String, metaData: [String: Any])
+    func processPushMetricsFromBGQ(token: String?, event: String, deliveryId: String, timestamp: String, metaData: [String: Any])
```

Comment:
```suggestion
    func processMetricsFromBGQ(token: String?, event: String, deliveryId: String, timestamp: String, metaData: [String: Any])
```

since you are planning on reusing for both push and in-app?

Reply:
Done in a new commit as it required changes in multiple files.